### PR TITLE
Use the new sort function in the planting sites local sort

### PR DESCRIPTION
**Pull request by nickgraz**

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2244?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2244** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2244?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2242** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2242?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2235** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2235?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2244?utm_source=stack-comment).

File: src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx [hunk 1]
```diff
@@ -19,33 +19,7 @@ import PlantingSiteTypeSelect from 'src/scenes/PlantingSitesRouter/edit/Planting
 import { useOrganization, useTimeZones } from 'src/providers/hooks';
 import { setTimeZone, useDefaultTimeZone } from 'src/utils/useTimeZoneUtils';
 import useForm from 'src/utils/useForm';
-
-type SiteProperty = keyof PlantingSiteSearchResult;
-
-const getVal = (site: PlantingSiteSearchResult, key: SiteProperty) => {
-  if (key === 'numPlantingSubzones') {
-    return site['numPlantingSubzones(raw)'] ?? 0;
```
Comment (nickgraz) [inline src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx:27]: One thing to note, the `sortResults` function I made doesn't fall back to 0, is that OK here? Will the search API return a result without this key? Easy to modify if we should always fall back to 0.

Comment (karthikbtf) [reply src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx:27]: In this case the BE has undefined values and we want to fallback to 0. Thanks for checking 🙏

File: src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx [hunk 2]
```diff
@@ -108,10 +82,13 @@ export default function PlantingSitesList(): JSX.Element {
 
       if (featureFlagSites && sites.some((site) => site.isDraft)) {
         // sort merged results by sort order
-        const sortField: SiteProperty = (searchSortOrder?.field ?? 'name') as SiteProperty;
-        const isAscending = searchSortOrder?.direction === 'Ascending';
-
-        sites.sort(siteSortFunction(sortField, isAscending, activeLocale || undefined));
+        return sortResults(sites, activeLocale, searchSortOrder, [
```
Comment (karthikbtf) [inline src/scenes/PlantingSitesRouter/view/PlantingSitesList.tsx:85]: Wow 😮 this is such a nice cleanup!!

**Review by karthikbtf — APPROVED** [review]
Many thanks 💯

File: src/utils/searchAndSort.ts [hunk 3]
```diff
@@ -74,11 +74,12 @@ const searchConditionMet = <T extends Record<string, unknown>>(result: T, condit
 
 // Try to get the `*(raw)` field if it exists, otherwise fall back to the regular field
 const getRawField = <T extends Record<string, unknown>>(result: T, field: string): unknown | undefined => {
-  const rawField = result[`${field}(raw)`];
-  if (rawField !== undefined) {
-    return rawField;
+  const fallback = [result[`${field}(raw)`], result[field], 0];
+  for (const value of fallback) {
```
Comment (karthikbtf) [inline src/utils/searchAndSort.ts:78]: this looks good, will it affect the deliverables sort in any way?

Comment (nickgraz) [reply src/utils/searchAndSort.ts:78]: No, it should hit the second fallback, I'll add a test to illustrate. I think eventually the raw field will be implemented for deliverables, specifically for `numDocuments`.

**Pull request merged**